camlibs/ptp2/ptp-pack.c

```diff
@@ -751,19 +761,56 @@ ptp_unpack_Sony_DPD (PTPParams *params, const unsigned char* data, PTPDeviceProp
 
 	ptp_debug (params, "prop 0x%04x, datatype 0x%04x, isEnabled %d getset %d", dpd->DevicePropCode, dpd->DataType, isenabled, dpd->GetSet);
 
-	switch (isenabled) {
-	case 0: /* grayed out */
-		dpd->GetSet = 0;	/* just to be safe */
-		break;
-	case 1: /* enabled */
-		/* enable for sony mode 2 - GetSet is 0 for many settings e.g. iso that *are* settable */
-		if (params->sony_mode_ver==2) {
-			dpd->GetSet = 1;
-		} /* with sony mode 3 - GetSet is set correctly initially and doesn't need to be set here */
-		break;
-	case 2: /* display only */
-		dpd->GetSet = 0;	/* just to be safe */
-		break;
+	if (params->sony_mode_ver==2) {
+		/* Old mode */
+		if (dpd->GetSet & 0x80) {
+			/* This is a control value and can be set */
+			/* 0x81 - button, 0x83 - lock, 0x82 - notch, 0x84 - variable */
+			dpd->GetSet = PTP_DPGS_GetSet;
```

Comment:
this is falling through isenabled handling ... should there be an else or goto out or something?

Reply:
Ah ok, maybe... I guess some  'controls' can be (legitimately) disabled, like if the exposure mode precludes that action. For now, I'll add an else - since this needs to be tested per control.